public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts

```diff
@@ -0,0 +1,61 @@
+import { SceneGridItem, SceneGridLayout, VizPanel } from '@grafana/scenes';
+import { Dashboard, defaultDashboard, FieldConfigSource, Panel } from '@grafana/schema';
+import { sortedDeepCloneWithoutNulls } from 'app/core/utils/object';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { getPanelIdForVizPanelKey } from '../utils/utils';
+
+export function transformSceneToSaveModel(scene: DashboardScene): Dashboard {
+  const state = scene.state;
+  const timeRange = state.$timeRange!.state;
+  const body = state.body;
+  const panels: Panel[] = [];
+
+  if (body instanceof SceneGridLayout) {
+    for (const child of body.state.children) {
+      if (child instanceof SceneGridItem) {
+        panels.push(gridItemToPanel(child));
+      }
+    }
+  }
+
+  const dashboard: Dashboard = {
+    ...defaultDashboard,
+    title: state.title,
+    uid: state.uid,
+    time: {
+      from: timeRange.from,
+      to: timeRange.to,
+    },
+    panels,
+  };
+
+  // Has to be a cleaner and faster way to do this. Code copied from old SaveDashboardDrawer to get rid of undefined
+  const withoutUndefined = JSON.parse(JSON.stringify(dashboard, null, 2));
```

Comment:
ouch, yeah this is a lot of processing. Gues sth like this would work just fine:

```tsx
function removeUndefinedValues<T extends Record<string, any>>(obj: T): void {
  for (const key in obj) {
    if (obj.hasOwnProperty(key) && obj[key] === undefined) {
      delete obj[key];
    }
  }
}
```

Reply:
I removed this for now, the call to sortedDeepCloneWithoutNulls removes undefined as well, I think the only thing the to JSON and back does is maybe transform -Infinity to null , which sortedDeepCloneWithoutNulls then removes. If this is the case we can easily add that behavior to sortedDeepCloneWithoutNulls as well